xrootd/protocol/protocol/protocol.go

```diff
@@ -7,134 +7,234 @@
+func (resp *Response) IsMeta() bool {
+	return resp.flags&isMeta != 0
+}
 
-// SecurityOverride is an alteration needed to the specified predefined security level.
-// It consists of the request index and the security requirement the associated request should have.
-// Request index is calculated as:
-//     (request code) - (request code of Auth request)
-// according to xrootd protocol specification.
-type SecurityOverride struct {
-	RequestIndex byte
-	RequestLevel RequestLevel
+// SetIsMeta sets whether this server has meta attribute.
+func (resp *Response) SetIsMeta(v bool) {
+	if v {
+		resp.flags |= isMeta
+	} else {
+		resp.flags &^= isMeta
+	}
 }
 
-// RequestOptions specifies what should be returned as part of response.
-type RequestOptions byte
+// IsProxy indicates whether this server has proxy attribute.
+func (resp *Response) IsProxy() bool {
+	return resp.flags&isProxy != 0
+}
 
-const (
-	// RequestOptionsNone specifies that only general response should be returned.
-	RequestOptionsNone RequestOptions = 0
-	// ReturnSecurityRequirements specifies that security requirements should be returned
-	// if that's supported by the server.
-	ReturnSecurityRequirements RequestOptions = 1
-)
+// SetIsProxy sets whether this server has proxy attribute.
+func (resp *Response) SetIsProxy(v bool) {
+	if v {
+		resp.flags |= isProxy
+	} else {
+		resp.flags &^= isProxy
+	}
+}
 
-// Request holds protocol request parameters.
-type Request struct {
-	ClientProtocolVersion int32
-	Options               RequestOptions
-	// FIXME: Rename Reserved* fields to _ when automatically generated (un)marshalling will be available.
-	Reserved1 [11]byte
-	Reserved2 int32
+// IsSupervisor indicates whether this server has supervisor attribute.
+func (resp *Response) IsSupervisor() bool {
+	return resp.flags&isSupervisor != 0
 }
 
-// NewRequest forms a Request according to provided parameters.
-func NewRequest(protocolVersion int32, withSecurityRequirements bool) Request {
-	var options = RequestOptionsNone
-	if withSecurityRequirements {
-		options |= ReturnSecurityRequirements
+// SetIsSupervisor sets whether this server has supervisor attribute.
+func (resp *Response) SetIsSupervisor(v bool) {
+	if v {
+		resp.flags |= isSupervisor
+	} else {
+		resp.flags &^= isSupervisor
+	}
+}
+
+// ForceSecurity indicates whether signing is required even if the authentication
+// protocol does not support generic encryption.
+func (resp *Response) ForceSecurity() bool {
+	return resp.SecurityOptions&forceSecurity != 0
+}
+
+// SetForceSecurity sets whether signing is forced.
+func (resp *Response) SetForceSecurity(v bool) {
+	if v {
+		resp.SecurityOptions |= forceSecurity
+	} else {
+		resp.SecurityOptions &^= forceSecurity
+	}
+}
+
+// MarshalXrd implements xrootd/protocol.Marshaler
+func (o Response) MarshalXrd() (data []byte, err error) {
+	var enc xrdenc.Encoder
+	enc.WriteI32(o.BinaryProtocolVersion)
+	enc.WriteI32(int32(o.flags))
```

Comment:
hum... why isn't the marshaling handling the optional writing part, symmetric to what is done on the unmarshaling side?